packages/teraslice-state-storage/src/cached-state-storage/index.ts

```diff
@@ -0,0 +1,70 @@
+
+import LRU from 'lru-cache';
+import { DataEntity, TSError } from '@terascope/job-components';
+import { CacheConfig, MGetCacheResponse } from '../interfaces';
+
+export default class CachedStateStorage {
+    protected IDField: string;
+    private cache: LRU<string, DataEntity>;
+
+    constructor(config: CacheConfig) {
+        this.IDField = config.id_field;
+        this.cache = new LRU({
+            max: config.cache_size,
+            maxAge: config.max_age
+        });
+    }
+
+    private getIdentifier(doc: DataEntity) {
+        const id =  doc.getMetadata(this.IDField);
+        if (id == null) throw new TSError(`There is no field "${this.IDField}" set in the metadata for doc: ${JSON.stringify(doc)}`);
```

Comment:
This should also catch empty strings and we should avoid stringifying the doc and exposing too much info (use the TSError context).
```suggestion
        if (!id) throw new TSError(`There is no field "${this.IDField}" set in the metadata`, {
            context: { doc }
        });
```